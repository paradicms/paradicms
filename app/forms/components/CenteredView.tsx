import React from "react";
import {StyleSheet, View} from "react-native";

export const CenteredView: React.FunctionComponent<React.PropsWithChildren<{}>> = ({
  children,
}) => <View style={styles.container}>{children}</View>;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
