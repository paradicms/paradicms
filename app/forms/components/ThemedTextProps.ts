import {Text} from "react-native";
import {ThemeProps} from "./ThemeProps";

export type ThemedTextProps = ThemeProps & Text["props"];
