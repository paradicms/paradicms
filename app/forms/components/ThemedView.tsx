import * as React from "react";
import {View as DefaultView} from "react-native";
import {useThemeColor} from "../hooks/useThemeColor";
import {ThemeProps} from "./ThemeProps";

type ViewProps = ThemeProps & DefaultView["props"];

export const ThemedView: React.FunctionComponent<ThemeProps &
  DefaultView["props"]> = ({style, lightColor, darkColor, ...otherProps}) => {
  const backgroundColor = useThemeColor(
    {light: lightColor, dark: darkColor},
    "background"
  );

  return <DefaultView style={[{backgroundColor}, style]} {...otherProps} />;
};
