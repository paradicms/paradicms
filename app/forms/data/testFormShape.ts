import {schema} from "@paradicms/vocabularies";
import {ShapesGraph} from "@paradicms/shacl";
import {FormShape} from "@paradicms/shacl-forms";
import {testShapesGraph} from "@paradicms/test";

export const testFormShape = new FormShape({
  id: "test",
  label: "Test form shape",
  nodeRdfTypes: [schema.Person],
  shapesGraph: ShapesGraph.fromDatasetCore(testShapesGraph),
});
