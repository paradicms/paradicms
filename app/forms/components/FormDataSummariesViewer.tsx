import * as React from "react";
import {CenteredView} from "./CenteredView";
import {FormDataSummary} from "@paradicms/shacl-forms";
import {FormModelSummariesViewer} from "@paradicms/react-native-shacl-forms";

export const FormDataSummariesViewer: React.FunctionComponent<{
  formDataSummaries: readonly FormDataSummary[];
  onSelectFormDataSummary: (formDataSummary: FormDataSummary) => void;
}> = ({formDataSummaries, onSelectFormDataSummary}) => (
  <CenteredView>
    <FormModelSummariesViewer
      formModelSummaries={formDataSummaries}
      onSelect={onSelectFormDataSummary}
    />
  </CenteredView>
);
