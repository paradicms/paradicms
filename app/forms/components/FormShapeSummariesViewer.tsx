import * as React from "react";
import {CenteredView} from "./CenteredView";
import {FormShapeSummary} from "@paradicms/shacl-forms";
import {FormModelSummariesViewer} from "@paradicms/react-native-shacl-forms";

export const FormShapeSummariesViewer: React.FunctionComponent<{
  formShapeSummaries: readonly FormShapeSummary[];
  onSelectFormShapeSummary: (formShapeSummary: FormShapeSummary) => void;
}> = ({formShapeSummaries, onSelectFormShapeSummary}) => (
  <CenteredView>
    <FormModelSummariesViewer
      formModelSummaries={formShapeSummaries}
      onSelect={onSelectFormShapeSummary}
    />
  </CenteredView>
);
