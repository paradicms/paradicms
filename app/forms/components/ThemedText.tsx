import * as React from "react";
import {Text as DefaultText} from "react-native";
import {ThemedTextProps} from "./ThemedTextProps";
import {useThemeColor} from "../hooks/useThemeColor";

export const ThemedText: React.FunctionComponent<ThemedTextProps> = ({
  style,
  lightColor,
  darkColor,
  ...otherProps
}) => {
  const color = useThemeColor({light: lightColor, dark: darkColor}, "text");
  return <DefaultText style={[{color}, style]} {...otherProps} />;
};
