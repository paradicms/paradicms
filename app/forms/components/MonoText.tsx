import React from "react";
import {ThemedTextProps} from "./ThemedTextProps";
import {ThemedText} from "./ThemedText";

export const MonoText: React.FunctionComponent<ThemedTextProps> = props => (
  <ThemedText {...props} style={[props.style, {fontFamily: "space-mono"}]} />
);
