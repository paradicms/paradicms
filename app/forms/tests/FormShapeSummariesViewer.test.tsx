import * as React from "react";
import * as renderer from "react-test-renderer";
import {testFormShape} from "../data/testFormShape";
import {FormShapeSummariesViewer} from "../components/FormShapeSummariesViewer";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <FormShapeSummariesViewer
        formShapeSummaries={[testFormShape.summary]}
        onSelectFormShapeSummary={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
