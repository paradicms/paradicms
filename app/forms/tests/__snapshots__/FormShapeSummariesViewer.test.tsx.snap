// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#bcbbc1",
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="RNE__LISTITEM__padView"
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 2,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 2,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                  "paddingHorizontal": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 18,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Test form shape
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
